src/nunit.analyzers/DisposeFieldsInTearDown/DisposeFieldsInTearDownAnalyzer.cs

```diff
@@ -0,0 +1,442 @@
+
+        private static HashSet<string> AssignedIn(SemanticModel model, INamedTypeSymbol type, HashSet<string> symbols, SyntaxList<StatementSyntax> statements)
+        {
+            var assignedSymbols = new HashSet<string>();
+
+            foreach (var statement in statements)
+            {
+                HashSet<string> assignedSymbolsInStatement = AssignedIn(model, type, symbols, statement);
+                assignedSymbols.UnionWith(assignedSymbolsInStatement);
+            }
+
+            return assignedSymbols;
+        }
+
+        #endregion
+
+        #region DisposedIn
+
+        private static HashSet<string> DisposedIn(SemanticModel model, INamedTypeSymbol type, HashSet<string> symbols, IEnumerable<IMethodSymbol> methods)
+        {
+            var disposedSymbols = new HashSet<string>();
+
+            foreach (var method in methods)
+            {
+                HashSet<string> disposedSymbolsInMethod = DisposedIn(model, type, symbols, method);
+                disposedSymbols.UnionWith(disposedSymbolsInMethod);
+            }
+
+            return disposedSymbols;
+        }
+
+        /// <summary>
+        /// Returns a hash set of the fields disposed in <paramref name="symbol"/>.
+        /// </summary>
+        /// <param name="symbol">The method to look for.</param>
+        /// <param name="symbols">The symbols to check for assignment.</param>
+        /// <returns>HashSet of <paramref name="symbols"/> that are disposed in <paramref name="symbol"/>.</returns>
+        private static HashSet<string> DisposedIn(SemanticModel model, INamedTypeSymbol type, HashSet<string> symbols, IMethodSymbol symbol)
+        {
+            MethodDeclarationSyntax? method =
+                symbol.DeclaringSyntaxReferences.FirstOrDefault()?.GetSyntax() as MethodDeclarationSyntax;
+
+            return method is null ? new HashSet<string>() : DisposedIn(model, type, symbols, method);
+        }
+
+        private static HashSet<string> DisposedIn(SemanticModel model, INamedTypeSymbol type, HashSet<string> symbols, MethodDeclarationSyntax method)
+        {
+            if (method.ExpressionBody is not null)
+            {
+                return DisposedIn(model, type, symbols, method.ExpressionBody.Expression);
+            }
+
+            if (method.Body is not null)
+            {
+                return DisposedIn(model, type, symbols, method.Body);
+            }
+
+            return new HashSet<string>();
+        }
+
+        private static HashSet<string> DisposedIn(SemanticModel model, INamedTypeSymbol type, HashSet<string> symbols, ExpressionSyntax expression)
+        {
+            var disposedSymbols = new HashSet<string>();
+            if (expression is InvocationExpressionSyntax invocationExpression)
+            {
+                if (invocationExpression.Expression is MemberAccessExpressionSyntax memberAccessExpression &&
+                    memberAccessExpression.Expression is not ThisExpressionSyntax)
```

Comment:
Why is this condition necessary? As far as I can tell all test pass without this check

Reply:
I hope indeed nothing calls `this.Dispose()` in an NUnit test ... or in any other code for that matter.
Removed.